### PR TITLE
http: remove a sprintf call from the hot path

### PR DESCRIPTION
**Pull request by michoecho**
sprintf() is a slow way to convert integers to text. Replace it with something faster.
This saves a microsecond in the hot path.

### CODE REVIEW

File: src/http/httpd.cc [hunk 1]
```diff
@@ -121,7 +128,7 @@ future<> connection::start_response() {
         });
     }
     set_headers(*_resp);
-    _resp->_headers["Content-Length"] = to_sstring(
+    _resp->_headers["Content-Length"] = size_to_sstring(
```
Comment (tchaikov) [inline src/http/httpd.cc:131]: instead turning away from `to_sstring()` and creating `size_to_sstring()` as a more performant alternative, IMHO, i think a better way to address this issue is to fix `to_sstring()`.

see #1398

Comment (michoecho) [reply src/http/httpd.cc:131]: I'm not sure how `fmt` works. From https://github.com/scylladb/seastar/pull/1398:

```
template <typename string_type, typename T>
string_type to_sstring(T value) {
    auto size = fmt::formatted_size("{}", value);
    auto formatted = uninitialized_string<string_type>(size);
    fmt::format_to(std::back_inserter(formatted), "{}", value);
    return formatted;
```

Will this really compile to something as fast as `size_to_sstring`?

Comment (michoecho) [reply src/http/httpd.cc:131]: Well, apparently it should be in the ballpark: https://www.zverovich.net/2020/06/13/fast-int-to-string-revisited.html

Comment (michoecho) [reply src/http/httpd.cc:131]: Alright, I will close the PR for now. Maybe I'll compare the two when #1398 is merged.

**Pull request closed without merging**